modules/instrument_manager/jsx/instrumentManagerIndex.js

```diff
@@ -103,11 +103,14 @@ class InstrumentManagerIndex extends Component {
 
     const tabs = [
       {id: 'browse', label: 'Browse'},
-      {id: 'upload', label: 'Upload'},
     ];
 
+    if (this.props.hasPermission('instrument_manager_write')) {
+      tabs.push({id: 'upload', label: 'Upload'});
+    }
+
     const feedback = () => {
-      if (!this.state.data.caninstall) {
+      if (!this.state.data.caninstall && this.props.hasPermission('instrument_manager_write')) {
```

Comment:
out of curiousity is there a reason to do it this way instead of incorporating it into the "caninstall"?

Reply:
I didn't want to put it in the function `isQuatUserConfigured` which is used for `caninstall` in instrument_manager.class.inc because it seemed specific to whether the user is configured properly in the sql DB. But I could put it in that function if you think it's better, or add it to line 322 in instrument_manager.class.inc where caninstall is set.

Reply:
I would have probably put it in toArray where caninstall is set to avoid needing to do permission checks in javascript, but it's fine the way it is since it looks like you need to do some js side checks anyways..